core-war/src/main/webapp/wysiwyg/jsp/uploadWebsiteFile.jsp

```diff
@@ -63,22 +65,54 @@ response.setDateHeader ("Expires",-1); //prevents caching at the proxy server
-	    if (fileItem != null)
-	    {
-			String fichierName = FileUploadUtil.getFileName(fileItem);
-			File fichier = new File(thePath, fichierName);
-			FileUploadUtil.saveToFile(fichier, fileItem);
-
-			String urlPath = thePath.substring(thePath.indexOf("/website"));
-
-			%>
-			<script language="javascript">
-				window.opener.document.getElementById('txtUrl').value='<%=WebEncodeHelper.javaStringToJsString(urlPath+'/'+fichierName)%>';
-				window.close();
-			</script>
-			<%
-	    }
+        FileItem fileItem = httpRequest.getSingleFile();
+        if (fileItem != null)
+        {
+            try {
+                // Validate the path to prevent path traversal attacks
+                if (thePath == null || !thePath.contains("/website")) {
+                    throw new SecurityException("Invalid path: path must contain '/website'");
+                }
+                
+                // Get the website repository from WysiwygController
+                String websiteRepository = WysiwygController.getWebsiteRepository();
+                
+                // Ensure the path is within the allowed website repository
+                File allowedRoot = new File(websiteRepository);
+                File requestedPath = new File(thePath);
+                
+                // Check if the requested path is within the allowed website repository
+                if (!requestedPath.getCanonicalPath().startsWith(allowedRoot.getCanonicalPath())) {
+                    throw new SecurityException("Security violation: attempted path traversal");
+                }
+                
+                // Process the upload with the validated path
+                String fichierName = FileUploadUtil.getFileName(fileItem);
+                File fichier = new File(requestedPath, fichierName);
+                
+                // Ensure the final destination is still within the allowed website repository
+                if (!fichier.getCanonicalPath().startsWith(allowedRoot.getCanonicalPath())) {
+                    throw new SecurityException("Security violation: destination file outside allowed path");
+                }
+                
+                FileUploadUtil.saveToFile(fichier, fileItem);
+                
+                String urlPath = thePath.substring(thePath.indexOf("/website"));
+                %>
+                <script language="javascript">
+                    window.opener.document.getElementById('txtUrl').value='<%=WebEncodeHelper.javaStringToJsString(urlPath+'/'+fichierName)%>';
+                    window.close();
+                </script>
+                <%
+            } catch (Exception e) {
+                SilverLogger.getLogger("wysiwyg").error("Error during file upload", e);
+                %>
+                <script language="javascript">
+                    alert("<%=WebEncodeHelper.javaStringToJsString(message.getString("Error") + ": " + e.getMessage())%>");
```

Comment:
Instead of `alert`, we have our own way to inform the user by using our popin mechanism. In this case, the alert statement should be replaced by the following one:
```javascript
sp.popup.error("<%=WebEncodeHelper.javaStringToJsString(message.getString("Error") + ": " + e.getMessage())%>");
```